Sources/SKCore/BuildSystem.swift

```diff
@@ -15,13 +15,13 @@ import Basic
 
 /// Provider of FileBuildSettings and other build-related information.
 ///
-/// The primary role of the build system is to answer queries for FileBuildSettings and (TODO) to
-/// notify clients when they change. The BuildSystem is also the source of related informatino,
-/// such as where the index datastore is located.
+/// The primary role of the build system is to answer queries for FileBuildSettings and to notify
+/// its delegate when they change. The BuildSystem is also the source of related information, such
+/// as where the index datastore is located.
 ///
 /// For example, a SwiftPMWorkspace provides compiler arguments for the files contained in a
 /// SwiftPM package root directory.
-public protocol BuildSystem {
+public protocol BuildSystem: AnyObject {
```

Comment:
Why do we need `AnyObject` now?

Reply:
Otherwise we need to make references to it mutable to be able to set it's delegate, I can change that if you'd like